src/lando/main/models/landing_job.py

```diff
@@ -64,23 +64,58 @@ def landed_revisions(self) -> dict:
         )
         return dict(revision_landing_jobs)
 
+    @property
+    def landed_revisions(self) -> dict:
+        """Return a mapping associating internal revision IDs with the ID of the landed Diff.
+
+        Note that the concept of a landed Diff is only relevant to Phabricator
+        transplants, and may therefore be None for other types of revisions.
+
+        """
+        revision_ids = [revision.id for revision in self.unsorted_revisions.all()]
+        revision_landing_jobs = (
+            RevisionLandingJob.objects.filter(
+                revision__id__in=revision_ids,
+                landing_job=self,
+            )
+            .order_by("index")
+            .values_list("revision_id", "diff_id")
+        )
+        return dict(revision_landing_jobs)
+
     @property
     def serialized_landing_path(self) -> list[dict]:
-        """Return landing path based on associated revisions or legacy fields."""
-        if self.unsorted_revisions:
+        """Return landing path based on associated revisions or legacy fields.
+
+        If the first revision in the list has a Phabricator revision ID, the whole stack
+        is assumed to be a Phabricator stack, and those IDs are returned. Otherwise,
+        internal IDs are returned as the revision_id.
+        """
+        if not self.unsorted_revisions:
+            # Legacy revisions.
+            return [
+                {"revision_id": "D{}".format(r), "diff_id": self.revision_to_diff_id[r]}
+                for r in self.revision_order
+            ]
```

Comment:
I don't think this is relevant any more, since we do not have any legacy revisions in the new Lando. This is a legacy-legacy concept (i.e., from back when we had revisions/diffs stored in a JSONField, prior to creating the RevisionLandingJob association table). I think it can be safely removed. We can probably remove the actual fields in a separate bug.